visualizer.py

```diff
@@ -484,43 +460,59 @@ def start_webserver():
             # Saving
             if saving.is_recording:
                 if ledsettings.color_mode == "Multicolor":
-                    saving.add_track("note_on", original_note, velocity, midiports.last_activity,
+                    saving.add_track("note_on", msg.note, velocity, msg_timestamp,
                                      wc.rgb_to_hex((red, green, blue)))
                 else:
-                    saving.add_track("note_on", original_note, velocity, midiports.last_activity)
+                    saving.add_track("note_on", msg.note, velocity, msg_timestamp)
 
         # Midi control change event
-        else:
-            control = find_between(str(msg), "control=", " ")
-            value = find_between(str(msg), "value=", " ")
+        elif msg.type == "control_change":
+            control = msg.control
+            value = msg.value
+
+            # midi control 64 = sustain pedal
+            if control == 64:
```

Comment:
I see now the reason for variable change, but why limit it only to a sustain pedal?

Reply:
Maybe I misread intent?
The variable is only ever checked in fade processing, and specifically against "pedal_deadzone" so I assumed that meant sustain.
Hmm, if it's any control, then what about "modulation", "volume", "expression", etc., or any number of private-use CC's?

I was thinking about tracking CC changes in "midi_state" when designing Color Modes, but so far nothing is using it, so it remains unimplemented...

Reply:
I see, I thought you had something else in mind.  If it is currently used in only one scenario, the new name is even more fitting.